### PR TITLE
Remove IsStream import from network modules

### PR DESCRIPTION
**Pull request by rnjtranjan**
Fixes:https://github.com/composewell/streamly/issues/1841

### CODE REVIEW

**Review by adithyaov — CHANGES_REQUESTED** [review]
Added a review commit

File: src/Streamly/Internal/Network/Socket.hs [hunk 1]
```diff
@@ -467,13 +469,20 @@ read = UF.first defaultChunkSize readWith
...
+mapM'_ f = D.mapM_ f . S.toStreamD
+
 -- | Write a stream of arrays to a handle.
 --
 -- @since 0.7.0
 {-# INLINE putChunks #-}
 putChunks :: (MonadIO m, Unboxed a)
     => Socket -> Stream m (Array a) -> m ()
-putChunks h = S.mapM_ (liftIO . writeChunk h)
+putChunks h = mapM'_ (liftIO . writeChunk h)
```
Comment (adithyaov) [inline src/Streamly/Internal/Network/Socket.hs:485]: You can use `S.fold FL.drain $ S.mapM $ ...` and check the benchmarks.

Comment (harendra-kumar) [reply src/Streamly/Internal/Network/Socket.hs:485]: Use `Fold.drainBy`.

File: src/Streamly/Internal/Network/Inet/TCP.hs [hunk 2]
```diff
@@ -107,7 +107,6 @@ import Streamly.Internal.Control.Concurrent (MonadAsync)
 import Streamly.Internal.Control.ForkLifted (fork)
 import Streamly.Internal.Data.Array.Unboxed.Type (Array(..), writeNUnsafe)
 import Streamly.Internal.Data.Fold.Type (Fold(..))
-import Streamly.Internal.Data.Stream.IsStream.Type (IsStream)
 import Streamly.Internal.Data.Stream (Stream)
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Network/Inet/TCP.hs:110]: We should import it from exposed module.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Remove the `import qualified Streamly.Internal.Data.Stream.IsStream as S` line from TCP module.

File: src/Streamly/Internal/Network/Socket.hs [hunk 3]
```diff
@@ -108,9 +106,13 @@ import qualified Streamly.Internal.Data.Array.Unboxed.Type as A
...
     (Array(..), newPinnedArrayBytes, asPtrUnsafe)
 import qualified Streamly.Internal.Data.Array.Stream.Foreign as AS
-import qualified Streamly.Internal.Data.Stream.IsStream as S
-import qualified Streamly.Internal.Data.Stream.StreamD.Type as D
-    (Stream(..), Step(..))
+import qualified Streamly.Internal.Data.Stream as S
+-- XXX Use explicit qualified imports here if the imports are not too many.
+-- XXX Eg.
+-- import qualified Streamly.Internal.Data.Stream.StreamD as D
+--     (Stream(..), Step(..))
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Network/Socket.hs:113]: Remove these comments. Are these addressed?

File: src/Streamly/Internal/Network/Socket.hs [hunk 4]
```diff
@@ -467,13 +469,20 @@ read = UF.first defaultChunkSize readWith
 -- Writing
 -------------------------------------------------------------------------------
 
+-- XXX We should not define it here. This module should not decide whether we
+-- use D or K. Maintanince will become tiresome. We should have it as
+-- Stream.Eliminate.mapM_
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Network/Socket.hs:474]: Remove these comments.

File: src/Streamly/Internal/Network/Socket.hs [hunk 5]
```diff
@@ -467,13 +469,20 @@ read = UF.first defaultChunkSize readWith
...
+{-# INLINE mapM'_ #-}
+mapM'_ :: Monad m => (a -> m b) -> S.Stream m a -> m ()
+mapM'_ f = S.fold (FL.drainBy f)
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Network/Socket.hs:477]: Use this directly.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Pull request merged**